### DISCUSSION TITLE
gridpoints forecast api returning old data

### DISCUSSION THREAD

**Discussion by dylex** [weather-gov/api#513] [General]
Over the last week or so, requests like https://api.weather.gov/gridpoints/OKX/34,33 have been sometimes returning forecast data from 2021-12-31, covering through 2022-01-07:
```
        "updateTime": "2021-12-31T11:18:10+00:00",
        "validTimes": "2021-12-31T05:00:00+00:00/P7DT8H",
```
This problem seems to persist for a few hours, then go away again, returning correct data for the current time.  This may be some sort of caching issue, because adding query parameters like `?debug=1` sometimes fixes it.

**Comment by JJKraw** [comment]
Search the rest of this forum. This is a well-known problem.

**Comment by michaelstringer** [comment]
Yeah, it's really sad, @dylex. We finally threw our hands up and ditched the API feed. When it works, it's great, but we really strive to provide consistent, updated info on our website. We were showing week-old forecast data this week so my boss made me pull the plug. Fingers crossed this gets fixed soon.

**Comment by dylex** [comment]
Ah, I see, like #329. Since this problem only started for me and seemed to correspond to 2022-01-01 I assumed it was new so checked the recent posts but didn't see anything.  Thanks for the pointer.  FWIW, so far adding `?debug=0` has fixed this for me.  I wonder if adding some sort of unique query string or header to each request somehow would be a way to work around it.

**Comment by JJKraw** [comment]
What has worked for me (caveat - I'm only getting the data for one location for a web site)

- I check the forecast every 15 minutes (cron job)
- Data is cached so that site visitors don't have to wait for the fetches
- Upon receiving data that is fresher than my cached data, I update the cache
- If the received data is old, I ignore it

I get enough correct hits on any given day to keep the cache fresh.  Sometimes gets one forecast cycle behind, but usually not.
